src/Functions/FunctionsConversion.h

```diff
@@ -387,125 +394,206 @@ struct ConvertImpl
 
 /** Conversion of DateTime to Date: throw off time component.
   */
-template <typename Name> struct ConvertImpl<DataTypeDateTime, DataTypeDate, Name, ConvertDefaultBehaviorTag>
-    : DateTimeTransformImpl<DataTypeDateTime, DataTypeDate, ToDateImpl> {};
+template <typename Name, DateTimeOverflowMode date_time_overflow_mode>
+struct ConvertImpl<DataTypeDateTime, DataTypeDate, Name, ConvertDefaultBehaviorTag, date_time_overflow_mode>
+    : DateTimeTransformImpl<DataTypeDateTime, DataTypeDate, ToDateImpl<date_time_overflow_mode>, false> {};
 
 /** Conversion of DateTime to Date32: throw off time component.
   */
-template <typename Name> struct ConvertImpl<DataTypeDateTime, DataTypeDate32, Name, ConvertDefaultBehaviorTag>
-    : DateTimeTransformImpl<DataTypeDateTime, DataTypeDate32, ToDate32Impl> {};
+template <typename Name, DateTimeOverflowMode date_time_overflow_mode>
+struct ConvertImpl<DataTypeDateTime, DataTypeDate32, Name, ConvertDefaultBehaviorTag, date_time_overflow_mode>
+    : DateTimeTransformImpl<DataTypeDateTime, DataTypeDate32, ToDate32Impl<date_time_overflow_mode>, false> {};
 
 /** Conversion of Date to DateTime: adding 00:00:00 time component.
   */
+template <DateTimeOverflowMode date_time_overflow_mode = default_date_time_overflow_mode>
```

Comment:
(all of the To<something> structs actually belong into DateTimeTransforms.h)

Reply:
This is true. IDK what was the reason for splitting this stuff between two files. I can actually move them all to DateTimeTransforms.h, but I'd prefer keeping this structure in this PR and making another PR with these changes

Reply:
Sure. Could be a follow-up PR or so.